NethermindEth/teeception
# Twitter and Proton encumbrance

## Description
This PR finishes the first implementation of encumbrance for Twitter and Proton. There were quite some missing pieces in the current draft implementation, such as supporting suspicious login flows on Twitter and the OAuth token fetching. This was also used as an opportunity to refactor the package structure of these utilities.
We are still missing verification email removal on Proton, which should be added as an issue.

## Type of change


## How Has This Been Tested?
1. Create a Proton account with a verifiation email (to avoid spam flagging)
2. Wait 1h
3. Create a Twitter account
4. Get the developer credentials for the Twitter account, and also register the OAuth read-write permissions and callback URL in the app settings (callback URL should be `http://IP:PORT/callback`)
5. Set the relevant env variables
6. `export $(cat .env | xargs) && go run cmd/setup/main.go`

## Checklist:
